### PR TITLE
report: alias element.analyses

### PR DESCRIPTION
**Pull request by PiTrem**
Report function expecting  sample.analyses 
but this data is now only under sample.containers

### CODE REVIEW

File: app/packs/src/apps/mydb/elements/details/reports/SectionReaction.js [hunk 1]
```diff
@@ -11,11 +11,7 @@ const SectionReaction = ({reaction, settings, configs}) => {
          tlc_solvents, rf_value, status } = reaction;
   const dangerousProducts = dangerous_products;
 
-  const has_analyses = products.map( sample => {
-    if(sample.analyses.length != 0) {
-      return true;
-    }
-  }).filter(r => r!=null).length != 0;
+  const has_analyses = !!(products.find((sample) => sample.analysesPresent()));
```
Comment (FabianMauz) [inline app/packs/src/apps/mydb/elements/details/reports/SectionReaction.js:14]: Why again the double exclamation marks ? 

sample.analysesPresent( ) already returns a boolean

File: app/packs/src/models/Element.js [hunk 2]
```diff
@@ -119,6 +119,27 @@ export default class Element {
...
+  // Return true if the element has at least one analysis
+  analysesPresent() {
+    if (!this.container) { return false; }
+
+    const analysesContainer = this.container.children?.find((container) => container?.container_type === 'analyses');
+    if (!analysesContainer) { return false; }
+
+    const analysis = analysesContainer.children?.find((container) => container?.container_type === 'analysis');
+    return !!analysis;
+  }
```
Comment (FabianMauz) [inline app/packs/src/models/Element.js:131]: could that function be shorten to
```suggestion
analysesPresent() {
   return analyses().length > 0
} 
```

**Review by FabianMauz — APPROVED** [review]
Some minor questions

File: app/packs/src/models/Element.js (hunk 2 above)
Comment (PiTrem) [reply app/packs/src/models/Element.js:131]: Indeed.
But I think it proposed change could be faster when the element has heaps of analyses :) 
Also  the functions  defined above that one could also have re-used, but I think they   they need a rewrite, so did not do that here.

**Pull request merged**